### PR TITLE
Feature - Export and Import tests and rawModeData to separate files

### PR DESCRIPTION
**Pull request by lorneb**
Hi new to github pull requests so go easy!.

This is a new feature that allows Newman to 
-J export to JSON the requets tests and rawMode fields to separate files, I found this handy when I need to edit the tests or data locally in an editor, bulk search and replace etc.  The import also creates a new collection file, it uses a convention of "file:XXXXX".

So after the export has been done, in the new collection file all the tests and rawModeData fields will be set to e.g for tests:   tests:"file:/somefolder/folderName.requestName.tests.json"

-K import

The reverse of the -J option, recombines the external separate tests and rawModeData back into the original postman format.  Also creates a new collection as output preserving original

Show folder name (if present) as test name in output,  The tests I made had several folders with the same name tests existing under different folders.  The test output was not including the folder name as part of the test name.

### CODE REVIEW

File: bin/newman [hunk 1]
```diff
@@ -33,6 +33,8 @@ function parseArguments() {
 		.option('-d, --data [file]', 'Specify a data file to use either json or csv', null)
 		.option('-g, --global [file]', 'Specify a Postman globals file [file]', null)
 		.option('-G, --exportGlobals [file]', 'Specify an output file to dump Globals before exiting [file]', null)
+		.option('-J, --exportJSON data to path [path]', 'Specify an output path to dump tests and rawModeData to [path]', null)
+        .option('-K, --importJSON data from path [path]', 'Specify an input path to import tests and rawModeData from [path]', null)
```
Comment (czardoz) [inline bin/newman]: Could you use tab indentation in this line as well?

File: bin/newman [hunk 2]
```diff
@@ -182,6 +184,14 @@ function main() {
 		newmanOptions.exportEnvironmentFile = program.exportEnvironment;
 	}
 
+    if (program.exportJSON) {
```
Comment (czardoz) [inline bin/newman]: Could you use tab indentation in this line?

File: bin/newman [hunk 3]
```diff
@@ -201,6 +211,8 @@ function main() {
 		try {
 			// read collection from the filesystem
 			collectionJson = JSON5.parse(fs.readFileSync(program.collection, 'utf8'));
+            //save the original file name if provided
+            newmanOptions.collectionFileName = program.collection;
```
Comment (czardoz) [inline bin/newman]: Could you use tab indentation in this line?

File: src/Newman.js [hunk 4]
```diff
@@ -25,6 +26,9 @@ var Newman = jsface.Class([Options, EventEmitter], {
      * @param {object} Newman options
      */
     execute: function(requestJSON, options, callback) {
+        
+        log.note("this is the new version");
```
Comment (czardoz) [inline src/Newman.js]: What is the purpose of this log statement?

File: src/Newman.js [hunk 5]
```diff
@@ -86,7 +90,15 @@ var Newman = jsface.Class([Options, EventEmitter], {
         // setup the iteration runner with requestJSON passed and options
         this.iterationRunner = new IterationRunner(requestJSON, this.getOptions());
 
-        this.iterationRunner.execute();
+        if (options.exportJSON) {
+            importer.exportJSON(requestJSON, options);
+        } 
+        else if (options.importJSON) {
```
Comment (czardoz) [inline src/Newman.js]: ``if`` should be on the next line

File: src/Newman.js [hunk 6]
```diff
@@ -86,7 +90,15 @@ var Newman = jsface.Class([Options, EventEmitter], {
...
+            importer.importJSON(requestJSON, options);
+        }else {
```
Comment (czardoz) [inline src/Newman.js]: could you move else to the next line too?

File: src/models/RequestModel.js [hunk 7]
```diff
@@ -1,5 +1,6 @@
 var jsface = require('jsface'),
-	ParentModel = require('./ParentModel.js');
+	ParentModel = require('./ParentModel.js'),
+    fs     = require('fs');
```
Comment (czardoz) [inline src/models/RequestModel.js]: Can you use tab indentation here?

**Comment by czardoz** [comment]
Thanks, @lorneb! 

I haven't completely looked at the code, only the first few files. (I'll do the rest today). Could you perhaps fix the code style issues that are making the build fail? https://travis-ci.org/postmanlabs/newman/jobs/83023252

**Comment by lorneb** [comment]
Hi sorry about the space / tab problems I was just using Brackets with the default settings,  what do you normally use for an editor?   I won't have a chance to make any changes until next Tuesday but I will try and sort them all out then.

File: src/Newman.js (hunk 4 above)
Comment (lorneb) [inline src/Newman.js]: I'll remove hangover from ensuring I was executing the correct code

**Comment by emilva** [comment]
Possible to feature creep something to this pull request?

I would like rename the exported rawModeData file into something more general, and include other properties like preRequestScript, data, url, method, pathVariables, headers, dataMode as I often edit these properties when editing/creating tests.

Also, it would be nice to be able to create tests that do not exist in a collection yet, that is: create new folder/test when importing. This would enable any developer to write postman tests in their fav editor.

**Comment by czardoz** [comment]
Closing this, because it cannot be merged into V3. It's possible to address this using custom newman reporters, which we'll document shortly :smile: 

Thanks a bunch for the PR, @lorneb!

**Pull request closed without merging**